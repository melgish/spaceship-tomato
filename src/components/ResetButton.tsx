import { ButtonHTMLAttributes } from 'react'

type Props = Readonly<Omit<ButtonHTMLAttributes<HTMLButtonElement>, 'type' | 'className'>>

export default function ResetButton({ children = 'Clear', ...props }: Props) {
  return (
    <button {...props} className="slate" type="button">
      {children}
    </button>
  )
}
