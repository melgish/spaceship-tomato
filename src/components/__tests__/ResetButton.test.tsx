import { expect, it } from 'vitest'
import { render } from '@testing-library/react'
import ResetButton from '@/components/ResetButton'

it('should render', () => {
  const { container } = render(<ResetButton />)

  expect(container).toMatchSnapshot()
})
