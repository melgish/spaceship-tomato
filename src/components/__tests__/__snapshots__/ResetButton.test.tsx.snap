// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render 1`] = `
<div>
  <button
    class="slate"
    type="button"
  >
    Clear
  </button>
</div>
`;
